File: src/10장/10-4.py
testcase = int(input())

for i in range(testcase):
    map = {}
    answer = 1
    n = int(input())
    for j in range(n):
        a, b = input().split()
        if not b in map:
            map[b] = 1
        else:
            map[b] += 1
    for k in map.keys():
        answer = answer * (map[k]+1)
    print(answer-1)

File: src/10장/10-4.java
import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int T = Integer.parseInt(br.readLine());
		while(T-->0){
			HashMap<String, Integer> map = new HashMap<>();
			int n = Integer.parseInt(br.readLine());
			for (int j = 0; j < n; j++) {
				String b = br.readLine().split(" ")[1];
				if (map.containsKey(b)) {
					map.put(b, map.get(b) + 1);
				} else {
					map.put(b, 1);
				}
			}
			int answer = 1;
			for (int y : map.values()) {
				answer *= y + 1;
			}
			System.out.println(answer - 1);
		}
	}
}